daemon/cluster/cluster.go

```diff
@@ -446,11 +446,20 @@ func (c *Cluster) UnlockSwarm(req types.UnlockRequest) error {
 
 	c.mu.RLock()
 	state := c.currentNodeState()
+	if !state.IsActiveManager() {
+		if err := c.errNoManager(state); err != errSwarmLocked {
+			c.mu.RUnlock()
+			return err
+		}
+	}
+
 	nr := c.nr
 	c.mu.RUnlock()
+
 	if nr == nil || errors.Cause(state.err) != errSwarmLocked {
```

Comment:
Actually I built the binary, and tested the case that daemon is a normal or active manager. And it returns `Error response from daemon: swarm is not locked`.

And I think the added test case `func (s *DockerSwarmSuite) TestUnlockEngineAndUnlockedSwarm(c *check.C)` also covered this.

To be honest I am a bit confused as well. Correct me if I missed something. @tonistiigi